programs/zk-token-proof/src/lib.rs

```diff
@@ -64,7 +64,7 @@ where
             return Err(InstructionError::InvalidAccountData);
         }
 
-        proof_context_account.set_data(context_state_data)?;
+        proof_context_account.set_data_from_slice(&context_state_data)?;
```

Comment:
given

```
impl<T: Pod> ProofContextState<T> {
    pub fn encode(
        context_state_authority: &Pubkey,
        proof_type: ProofType,
        proof_context: &T,
    ) -> Vec<u8> {
        let mut buf = Vec::with_capacity(size_of::<Self>());
        buf.extend_from_slice(context_state_authority.as_ref());
        buf.push(ToPrimitive::to_u8(&proof_type).unwrap());
        buf.extend_from_slice(bytes_of(proof_context));
        buf
    }
```

Shoudn't we make encode write directly into the account, skipping the temporary vec + another memcpy?

Reply:
heh, i wasn't bothered enough to erase the newly extra memcpy. ;) sounds nice suggestion. @samkim-crypto could you address this after this pr is merged? or, should i do that in this pr? (if could, i want to offload to you.. thanks in advance!)

Reply:
Oh yes, sorry I missed this. Yes, I can definitely address this on a follow-up!

Reply:
> Oh yes, sorry I missed this. Yes, I can definitely address this on a follow-up!

thanks very much! I just merged this.